File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l = 0, r = 1;
        int ans = 0;
        while (r < prices.size()){
            if (prices[l] < prices[r]){
                int curr = prices[r] - prices[l];
                ans = max(ans, curr);
            }
            else{
                l = r;
            }
            r++;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        f = defaultdict(int)

        l = 0
        max_freq = 0
        ans = 0

        for r in range(len(s)):
            f[s[r]] += 1

            max_freq = max(max_freq, f[s[r]])

            while (r-l+1) - max_freq > k:
                f[s[l]] -= 1
                l += 1
            
            ans = max(ans, r-l+1)
        
        return ans

